Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/AzureWebAppMvpModel.java

```diff
@@ -574,12 +577,45 @@ public void deleteDeploymentSlotNode(final String subscriptionId, final String a
         return webApps;
     }
 
+    /**
+     * List available web containers for jar files.
+     */
+    public List<WebAppUtils.WebContainerMod> listWebContainersForJarFile(JdkModel jdkModel) {
+        if (jdkModel == null) {
+            return Collections.emptyList();
+        }
+        final String javaVersion = jdkModel.getJavaVersion().toString();
+        if (javaVersion.startsWith("1.8")) {
+            return JAVA_8_JAR_CONTAINERS;
+        } else if (javaVersion.startsWith("11")) {
+            return JAVA_11_JAR_CONTAINERS;
+        }
+        return Collections.emptyList();
+    }
+
+
+    /**
+     * List available web containers for war files.
+     */
+    public List<WebAppUtils.WebContainerMod> listWebContainersForWarFile() {
+        return Arrays.asList(
+                WebAppUtils.WebContainerMod.Newest_Jetty_91,
+                WebAppUtils.WebContainerMod.Newest_Jetty_93,
+                WebAppUtils.WebContainerMod.Newest_Tomcat_70,
+                WebAppUtils.WebContainerMod.Newest_Tomcat_80,
+                WebAppUtils.WebContainerMod.Newest_Tomcat_85,
+                WebAppUtils.WebContainerMod.Newest_Tomcat_90
+        );
+    }
+
     /**
      * List available Web Containers.
      */
     public List<WebAppUtils.WebContainerMod> listWebContainers() {
         List<WebAppUtils.WebContainerMod> webContainers = new ArrayList<>();
         Collections.addAll(webContainers, WebAppUtils.WebContainerMod.values());
+        webContainers.remove(WebAppUtils.WebContainerMod.Java_SE_8);
```

Comment:
Can we use tricks to enumerate the WebAppUtils.WebContainerMod to exclude Java-SE*?
Otherwise when WebContainerMod  add new enum values like Java-SE13, your code will break.

Reply:
Sorry, this is a outdated function which is not used, instead we used listWebContainersForWarFile() and listWebContainersForWarFile(), has removed it and update the test cases.